maestro-cli/src/main/java/maestro/cli/runner/MaestroCommandRunner.kt

```diff
@@ -82,23 +83,39 @@ object MaestroCommandRunner {
                     success = false
                 },
             ).executeCommands(commands)
+            return success
         }
 
-        if (skipInitFlow) {
-            initCommands.forEach { initCommandStatuses[it] = CommandStatus.COMPLETED }
+        val initAppState: File? = if (initFlow == null) {
+            null
         } else {
-            executeCommands(initCommands, initCommandStatuses)
-        }
+            val state = if (cachedAppState == null) {
```

Comment:
How could we reuse that logic in our backend runner?

Reply:
As discussed offline - This logic is actually dependent on the context. Not much can be reused.